### PR TITLE
POST request functionality to Net::HTTP class and tests for .get() and .post() methods

### PR DESCRIPTION
**Pull request by ear7h**
http.go
	HTTP.post() - uses native go http.Post() method to post data via http
http_test.go tests for http class
	HTTP.get()
	HTTP.post()

### CODE REVIEW

File: vm/http_test.go [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+		Net::HTTP.get("http://127.0.0.1:3000")
+		`, "GET Hello World"},
+		{`
+		require "net/http"
+
+		Net::HTTP.post("http://127.0.0.1:3000", "text/plain", "Hi Again")
+		`, "POST Hi Again"},
+	}
+
+	c := make(chan bool, 1)
```
Comment (st0012) [inline vm/http_test.go]: I want to move the whole server setup part above the test cases.

File: vm/http_test.go [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+
+		m.HandleFunc("/", func (w http.ResponseWriter, r *http.Request) {
+			w.WriteHeader(http.StatusOK)
+
+			if r.Method == http.MethodPost {
+				b, err := ioutil.ReadAll(r.Body)
+				if err != nil {
+					panic(err)
+				}
+				w.Write([]byte(r.Method + " " + string(b)))
```
Comment (st0012) [inline vm/http_test.go]: I think just write `fmt.Sprintf("POST %s", string(b))` is better. Because we already know this method is surely `POST`

File: vm/http_test.go [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+			} else {
+				w.Write([]byte(r.Method + " Hello World"))
```
Comment (st0012) [inline vm/http_test.go]: Same as above

Comment (ear7h) [reply vm/http_test.go]: I think using r.Method on this line is preferable as it can be used to test DELETE, HEAD, etc.

File: vm/http_test.go (hunk 2 above)
Comment (ear7h) [reply vm/http_test.go]: That's true I'll change that

File: vm/http_test.go (hunk 3 above)
Comment (st0012) [reply vm/http_test.go]: I see, I didn't know you will also add those methods.

**Comment by ear7h** [comment]
Committed changes from your feedback

**Comment by st0012** [comment]
One more thing, can you also add tests for false usage and none successful responses?
Those should be placed at new test functions, you can see error tests at array.go. Find those function which's name contains "Fail".

**Comment by ear7h** [comment]
also changed the way url's are connected for get requests

File: vm/http.go [hunk 4]
```diff
@@ -54,23 +57,64 @@ func builtinHTTPClassMethods() []*BuiltInMethodObject {
...
+
+					body := args[2].(*StringObject).value
+
+					resp, err := http.Post(uri.String(), contentType, strings.NewReader(body))
 
 					if err != nil {
 						return t.vm.initErrorObject(InternalError, err.Error())
 					}
+					if resp.StatusCode != http.StatusOK {
+						return t.vm.initErrorObject(InternalError, resp.Status)
```
Comment (st0012) [inline vm/http.go]: BTW This error can be more specific, like `HTTPError`. You can create new error type in `error.go`. But let's merge this first.

**Comment by st0012** [comment]
Because codeclimate still haven't upgrade its `golint`, so I still need to skip the codeclimate check.
